### PR TITLE
Bugfix/drop old tables

### PR DESCRIPTION
**Pull request by bdmogal**
Jira: [CDAP-1802](https://issues.cask.co/browse/CDAP-1802)
Build: http://builds.cask.co/browse/CDAP-RBT191-1

### CODE REVIEW

File: cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java [hunk 1]
```diff
@@ -345,12 +344,16 @@ private Action parseAction(String action) {
 
   private void performUpgrade() throws Exception {
     LOG.info("Upgrading System and User Datasets ...");
+    HBaseAdmin hBaseAdmin = new HBaseAdmin(hConf);
     DatasetUpgrader dsUpgrade = injector.getInstance(DatasetUpgrader.class);
     dsUpgrade.upgrade();
+    hBaseTableUtil.dropTable(hBaseAdmin, dsUpgrade.getDatasetInstanceMDSUpgrader().getOldDatasetInstanceTableId());
```
Comment (anwar6953) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java]: if dropping tables right after each upgrade method, why not drop them within the Upgrader class?

File: cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java [hunk 2]
```diff
@@ -423,10 +427,7 @@ private void createNamespaces() {
       Throwables.propagate(e);
     }
     LOG.info("Creating and registering {} namespace", Constants.DEFAULT_NAMESPACE);
-    getStore().createNamespace(new NamespaceMeta.Builder()
-                                 .setName(Constants.DEFAULT_NAMESPACE)
-                                 .setDescription(Constants.DEFAULT_NAMESPACE)
-                                 .build());
+    getStore().createNamespace(Constants.DEFAULT_NAMESPACE_META);
```
Comment (anwar6953) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java]: i know this is from before, but did you come across why the metastore needs this in it?

File: cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java (hunk 1 above)
Comment (bdmogal) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java]: You need ``HBaseTableUtil``, ``HBaseAdmin``, ``Configuration`` objects to drop a table. I feel very strongly that classes such as ``FileMetadataManager`` should not know about these classes at all. 

If the Upgrader classes were subclasses of the ``AbstractUpgrader``, then I could have done by having these objects as protected members in ``AbstractUpgrader``. However, seeing that that is not possible under the current design, I found that this approach was a decent compromise for 2.8.0. 

We need to refactor this code to have the Upgrader classes obey a hierarchy eventually. Thoughts?

File: cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java (hunk 2 above)
Comment (bdmogal) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java]: Didn't investigate further after we discussed. @rohitsinha54 do you know why this is needed?

File: cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java (hunk 1 above)
Comment (anwar6953) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java]: Ok sounds good.

**Comment by anwar6953** [comment]
LGTM.

**Pull request merged**